### DISCUSSION TITLE
TTGO no longer builds correctly

### DISCUSSION THREAD

**Discussion by prschguy1** [PlummersSoftwareLLC/NightDriverStrip#245] [General]
Have been having some trouble with TTGO in recent builds. While it used to work correctly, it has had some issues for some time now. If I go back to earlier builds, it will build correctly and run just fine so it is unclear what has changed.. When I log the output I get this:

ELF file SHA256: 1650bd3f954e340d

E (1288) esp_core_dump_flash: Core dump flash config is corrupted! CRC=0x7bd5c66f instead of 0x0
Rebooting...
ets Jul 29 2019 12:21:46

rst:0xc (SW_CPU_RESET),boot:0x13 (SPI_FAST_FLASH_BOOT)
configsip: 0, SPIWP:0xee
clk_drv:0x00,q_drv:0x00,d_drv:0x00,cs0_drv:0x00,hd_drv:0x00,wp_drv:0x00
mode:DIO, clock div:2
load:0x3fff0030,len:1184
load:0x40078000,len:13192
load:0x40080400,len:3028
entry 0x400805e4
E (556) I2S: i2s_calculate_adc_dac_clock(1046): sample rate is too large
E (557) I2S: i2s_calculate_clock(1199): ADC clock calculate failed
E (561) esp_core_dump_flash:����ɕ�dump partition found!
E (563) esp_core_dump_flash: No core duReplacing Idle Tasks with TaskManager...
(I) (DebugLoopTaskEntry)(C0) >> DebugLoopTaskEntry
(I) 
E (121) gpio: GPIO can only be used as input mode
[   146][E][esp32-hal-gpio.c:130] __pinMode(): GPIO config failed
E (127) gpio: gpio_set_level(226): GPIO output gpio_num error
E (132) gpio: GPIO can only be used as input mode
[   160][E][esp32-hal-gpio.c:130] __pinMode(): GPIO config failed
E (142) gpio: gpio_set_level(226): GPIO output gpio_num error
(I) (setup)(C1) Initializing TFTSPI
(I) (setup)(C1) Could allocate 27 buffers but limiting it to 20
(I) 
(W) (setup)(C1) Reserving 20 LED buffers for a total of 46720 bytes...
(I) (setup)(C1) Adding LEDs to FastLED...
(I) (setup)(C1) Adding 768 LEDs to FastLED.
(I) (setup)(C1) Initializing effects manager...
(W) (InitEffectsManager)(C1) InitEffectsManager...
(W) (StartScreenThread)(C1) >> Launching Screen Thread
(I) (setup)(C1(Scrawing:
Entry)(C0) >> ScreenUpdateLoopEntry
(I) 
I) 
hing Drawing:
(E) (setup)(C1) Heap before launch: PASS
(W) (StartDrawThread)(C1) >> Launching Draw Thread
(W) (DrawLoopTaskEntry)(C1) >> DrawLoopTaskEntry
(W) 
(W) (Setup)(C1) >> GFX(C0) ------------(E) 
ateHandler)(C0) -------------------------------------------------------------------------------------
E) 
(W) u MerminateHandler)(C0) - NightDriverStrip Gur
W) u Meditation                              Unhandled Exception -
(E) (TerminateHandler)(C0) -------------------------------------------------------------------------------------
(I) (PrintOutputHeader)(C0) NightDriverStrip
(I) 
(I) (PrintOutputHeader)(C0) ------------------------------------------------------------------------------------------------------------
(I) (PrintOutputHeader)(C0) M5STICKC: 0, USE_M5DISPLAY: 0, USE_OLED: 0, USE_TFTSPI: 1, USE_LCD: 0, USE_AUDIO: 1, ENABLE_REMOTE: 1
(I) (PrintOutputHeader)(C0) Version 36: Wifi SSID: ................... - ESP32 Free Memory: 185304, PSRAM:0, PSRAM Free: 0
(I) (PrintOutputHeader)(C0) ESP32 Clock Freq : 240 MHz
(E) (TerminateHandler)(C0) ---------------------------------(W) (StartNetworkThread)(C1) >> Launching Networ
abort() was called at PC 0x40165ba4 on core 0

Looking at it I'm not sure whether it is a problem with GPIO or buffer rate is being over run by too many samples. Seems strange that it used to be ok, but now it is not. Would appreciate if any one has some ideas on what I need to change?

**Comment by davepl** [comment]
Looks like it might be running out of memory.  Find RESERVER_MEMORY and add 50,000 to it, which will reduce buffering for wifi.  Let me know if that helps!

- Dave

**Comment by prschguy1** [comment]
Hello Dave, thank you so much for jumping in and trying to help. Unfortunately, changing that value didn't help it. I had also tried changing max buffer value to no effect. Looks like whether I compile it myself, or use the web installer, the result is the same. I will look into it a bit more and see why it used to work, but now does not. With my limited understanding of coding, I'm a bit hung up on the statements about GPIO so will probably start looking there. I have tried toggling features on and off and that doesn't seem to effect it either. I have 3 boards and they all act the same, so I'm pretty sure it's not a board problem. Anyway, thanks so much for your thoughts. Really enjoy your videos and working with the code. Atom light is my favorite and built a really good-looking example of it. Trying to find a more capable esp32 to put in it at the moment. Thanks for all you do!
